lib/web_ui/lib/src/engine/dom_renderer.dart

```diff
@@ -458,22 +458,21 @@ $_glassPaneTagName * {
       ..bottom = '0'
       ..left = '0';
 
-    // Create a Shadow Root under the glass panel, and attach everything there,
-    // instead of directly underneath the glass panel.
-    final html.ShadowRoot glassPaneElementShadowRoot = glassPaneElement.attachShadow(<String, String>{
-      'mode': 'open',
-      'delegatesFocus': 'true',
-    });
+    // Create a [HostNode] under the glass pane element, and attach everything
+    // there, instead of directly underneath the glass panel.
+    final HostNode glassPaneElementShadowRoot = _createHostNode(glassPaneElement);
     _glassPaneShadow = glassPaneElementShadowRoot;
 
     bodyElement.append(glassPaneElement);
 
-    final html.StyleElement shadowRootStyleElement = html.StyleElement();
-    // The shadowRootStyleElement must be appended to the DOM, or its `sheet` will be null later...
-    glassPaneElementShadowRoot.append(shadowRootStyleElement);
+    if (glassPaneElementShadowRoot is ShadowDomHostNode) {
```

Comment:
This looks like an abstraction leak. Can the implementation of the host node attach the stylesheet?

Reply:
Let me see if I can reuse the `_applyCssRulesToSheet` somehow, when initializing the `ShadowDomHostNode`...

Reply:
If the applyCssRulesToSheet method continues growing, I think that we might need to figure out a way to clone the stylesheet from the body into the shadowDOM directly, so we don't make the method fatter than what it needs to be.